app/src/main/java/com/hedvig/app/feature/profile/data/ObserveProfileUiStateUseCase.kt

```diff
@@ -0,0 +1,24 @@
+package com.hedvig.app.feature.profile.data
+
+import arrow.core.Either
+import com.hedvig.android.owldroid.graphql.ProfileQuery
+import com.hedvig.app.feature.profile.ui.tab.ProfileQueryDataToProfileUiStateMapper
+import com.hedvig.app.feature.profile.ui.tab.ProfileUiState
+import com.hedvig.app.util.apollo.QueryResult
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapLatest
+
+class ObserveProfileUiStateUseCase(
+    private val profileRepository: ProfileRepository,
+    private val profileQueryDataToProfileUiStateMapper: ProfileQueryDataToProfileUiStateMapper,
+) {
+    suspend fun invoke(): Flow<Either<QueryResult.Error, ProfileUiState>> {
```

Comment:
Map to `ErrorMessage` here instead of `QueryResult.Error`

Reply:
Resolving due to this https://github.com/HedvigInsurance/android/pull/972